include/lsst/afw/image/PhotoCalib.h

```diff
@@ -451,6 +452,80 @@ class PhotoCalib : public table::io::PersistableFacade<PhotoCalib>, public table
 
     friend std::ostream &operator<<(std::ostream &os, PhotoCalib const &photoCalib);
 
+    /* Backwards compatibility with old Calib object */
+
+    /// No-op: for backwards compatibility with Calib.
+    [[deprecated("No-op: PhotoCalib never throws on negative instFlux. Will remove after v18.")]] static void
+    setThrowOnNegativeFlux(bool raiseException) noexcept {
+        ;  // do nothing!
+    }
+    [[deprecated("No-op: PhotoCalib never throws on negative instFlux. Will remove after v18.")]] static bool
+    getThrowOnNegativeFlux() noexcept {
+        return false;
+    }
+
+    /** @copydoc instFluxToMagnitude(double)const
+     * Deprecated: For backwards compatibility with Calib.
+     */
+    [
+            [deprecated("For backwards compatibility with Calib; use `instFluxToMagnitude` instead. To be "
+                        "removed after v18.")]] double
+    getMagnitude(double instFlux) const {
+        return instFluxToMagnitude(instFlux);
+    }
+
+    /** @copydoc instFluxToMagnitude(double,double)const
+     * Deprecated: For backwards compatibility with Calib.
+     */
+    [
+            [deprecated("For backwards compatibility with Calib; use `instFluxToMagnitude` instead. To be "
+                        "removed after v18.")]] ndarray::Array<double, 1>
+    getMagnitude(ndarray::Array<double const, 1> const &instFlux) const;
+    [
+            [deprecated("For backwards compatibility with Calib; use `instFluxToMagnitude` instead. To be "
+                        "removed after v18.")]] std::pair<double, double>
+    getMagnitude(double instFlux, double instFluxErr) const {
+        auto result = instFluxToMagnitude(instFlux, instFluxErr);
+        return std::make_pair<const double &, const double &>(result.value, result.error);
+    };
+    [[deprecated(
+            "For backwards compatibility with Calib; use `instFluxToMagnitude` instead. To be "
+            "removed after v18.")]] std::pair<ndarray::Array<double, 1>, ndarray::Array<double, 1>>
+    getMagnitude(ndarray::Array<double const, 1> const &instFlux,
+                 ndarray::Array<double const, 1> const &instFluxErr) const;
```

Comment:
Did the original C++ have a default of 0 for instFluxErr (as the Python does)? If so, please add it.

Reply:
I don't see where you see a default value: neither this new deprecated code, nor the original `Calib` object has any defaults for `instFluxErr`.

Were you instead thinking of `setFluxMag0()`? That one did have a 0 default for the error, which remains here at the python level, but seems silly at the C++ level since it the function just immediately throws.

Reply:
I guess I must have been thinking of setFluxMag0 and it looks like the C++ already has a default of 0, which is great.